tardis/energy_input/base.py

```diff
@@ -0,0 +1,369 @@
+import numpy as np
+import copy
+from tqdm.auto import tqdm
+import pandas as pd
+
+# from tardis.montecarlo.montecarlo_numba.r_packet import get_random_mu
+from tardis.energy_input.util import SphericalVector, GXPacket, GXPacketStatus
+from tardis.energy_input.gamma_ray_grid import (
+    distance_trace,
+    move_gamma_ray,
+    get_shell,
+    compute_required_packets_per_shell,
+)
+from tardis.energy_input.energy_source import (
+    setup_input_energy,
+    sample_energy_distribution,
+    intensity_ratio,
+)
+from tardis.energy_input.calculate_opacity import (
+    compton_opacity_calculation,
+    photoabsorption_opacity_calculation,
+    pair_creation_opacity_calculation,
+)
+from tardis.energy_input.gamma_ray_interactions import (
+    scatter_type,
+    compton_scatter,
+    pair_creation,
+    get_compton_angle,
+)
+from tardis.energy_input.util import (
+    get_random_theta_gamma_ray,
+    get_random_phi_gamma_ray,
+)
+from tardis import constants as const
+from astropy.coordinates import cartesian_to_spherical
+from tardis.montecarlo.montecarlo_numba.numba_config import CLOSE_LINE_THRESHOLD
+
+
+def initialize_packets(
+    number_of_shells,
+    packets_per_shell,
+    ejecta_volume,
+    inner_velocities,
+    outer_velocities,
+    decay_rad_db,
+):
+    """Initializes packet properties
+    and appends beta decay energy to output tables
+
+    Parameters
+    ----------
+    number_of_shells : int64
+        Number of shells in model
+    packets_per_shell : pandas dataframe
```

Comment:
for IDEs to pick-up on this stuff actually give them a valid python expression - `pandas.DataFrame`